Tools/clinic/clinic.py

```diff
@@ -792,11 +792,14 @@ def docstring_for_c_string(
             add('"')
         return ''.join(text)
 
-    def output_templates(self, f):
+    def output_templates(
+            self,
+            f: Function
+    ) -> dict[str, str]:
         parameters = list(f.parameters.values())
         assert parameters
-        assert isinstance(parameters[0].converter, self_converter)
-        del parameters[0]
+        first_param = parameters.pop(0)
+        assert isinstance(first_param.converter, self_converter)
```

Comment:
We could possibly code-golf this down a bit more:

```suggestion
        assert isinstance(parameters.pop(0).converter, self_converter)
```

Reply:
IMO, doing too many things in one line makes the code harder to read; I prefer the explicitness of the existing variant :)